### PR TITLE
Refactoring `Merkle_ledger`

### PR DESCRIPTION
**Pull request by rbonichon**
- activate warnings (and fix them)
- move interface definitions in `Intf` only (and thus avoid repetition)
- add interfaces to all modules
- remove some dead code (thanks to warnings too)

This is preliminary work/scaffolding for heavier changes (to come at some point down the road)

Still todo
- [x] Cleanup history (do not review until this is crossed)

### CODE REVIEW

**Comment by rbonichon** [comment]
!ci-build-me

File: src/lib/rocksdb/database.ml [hunk 1]
```diff
@@ -57,21 +57,19 @@ module Batch = struct
     Rocks.write t.db batch ; result
 end
 
-let copy _t = failwith "copy: not implemented"
-
 let remove t ~(key : Bigstring.t) : unit =
   Rocks.delete ?pos:None ?len:None ?opts:None t.db key
 
+let copy t : Bigstring.t =
```
Comment (mrmr1993) [inline src/lib/rocksdb/database.ml:63]: Aren't these changes already in the base branch (https://github.com/MinaProtocol/mina/pull/15199)? Same comments apply here as there, but I don't understand how they ended up in this PR at all

Comment (rbonichon) [reply src/lib/rocksdb/database.ml:63]: They shouldn't have. But the `intf` expansion depends on the existence of change in #15199, which, it appears were not properly segregated

**Comment by rbonichon** [comment]
!ci-build-me

**Review by mrmr1993 — APPROVED** [review]

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Pull request merged**